--- prelude-manager.conf.in.old	2007-05-08 20:48:32.000000000 +0200
+++ prelude-manager.conf.in	2007-05-08 20:50:59.000000000 +0200
@@ -115,7 +115,7 @@
 # [TextMod]
 #
 # logfile = stderr
-# logfile = @prefix@/var/log/prelude.log
+# logfile = @localstatedir@/log/prelude.log
 
 
 
@@ -141,7 +141,7 @@
 # format
 #
 # logfile = stderr
-# logfile = @prefix@/var/log/prelude-xml.log
+# logfile = @localstatedir@/log/prelude-xml.log
 
 
 
