--- mysql2pgsql.sh.old	2007-05-16 11:10:45.000000000 +0200
+++ mysql2pgsql.sh	2007-05-16 11:11:05.000000000 +0200
@@ -9,7 +9,7 @@
         -e 's/ TINYINT / INT2 /g' \
         -e 's/ SMALLINT UNSIGNED / INT8 /g' \
         -e 's/ SMALLINT / INT4 /g' \
-	-e 's/ BIGINT UNSIGNED / NUMERIC(20) /g' \
+	-e 's/ BIGINT UNSIGNED / INT8 /g' \
 	-e 's/ BIGINT / INT8 /g' \
 	-e 's/ INT\(EGER\)\{0,1\} UNSIGNED / INT8 /g' \
 	-e 's/ INT\(EGER\)\{0,1\} / INT4 /g' \
