--- libprelude-0.9.14-old/configure.in	2007-05-15 16:48:13.000000000 +0200
+++ libprelude-0.9.14/configure.in	2007-06-07 18:43:09.000000000 +0200
@@ -293,13 +293,13 @@
 
 if test x$have_system_ltdl != xyes; then
         with_ltdl_type="Convenience"
+        AC_CONFIG_SUBDIRS(libltdl)
 else
         with_ltdl_type="System"
         INCLTDL="-I$with_system_ltdl/include"
         LIBLTDL="-L$with_system_ltdl/lib -lltdl"
 fi
 
-AC_CONFIG_SUBDIRS(libltdl)
 AM_CONDITIONAL(HAVE_SYSTEM_LIBLTDL, test x$have_system_ltdl = xyes)
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
